huntsman/utils/light_control.py

```diff
@@ -0,0 +1,265 @@
+import qhue
+from os import path
+from qhue import Bridge, QhueException, create_new_username
+
+#note, light 5 can be 'uncommented' once installed, or causes problems using currently 
+
+
+#deal with error system - should errors be sent to pocs? how does pocs handle error messages?
+
+def manual_login_phillips_hue(hue_ip="10.88.21.10", verbose = False):
+    
+    """
+    A way to manually login to the hue bridge, requires the button on the bridge to be pushed.
+    Only useful should auto login fail and present at the dome.
+    
+    Inputs:
+        
+        hue_ip - the ip address of the hue bridge. This is is set to the default input "10.88.21.10" 
+    
+    """
+    
+    username = qhue.create_new_username(hue_ip)
+    
+    if verbose:
+    
+       print("The username created is", username)
+    
+    b = qhue.Bridge(hue_ip,username)
+    
+    return(b)
+    
+def automatic_login_phillips_hue(hue_ip = "10.88.21.10", file_path = "hue_username.txt", verbose = False):
+    
+    """
+    Automatic connection to the hue bridge using a stored username.
+    
+    Functions:
+        
+        Creates a text file with stored username upon first connection to the bridge, will
+        prompt the user to press the hue bridge button to connect for the fist time and writes
+        the file to the central directory. 
+        
+        Should a username already exist, the stored username is read and used to log into the
+        bridge automatically.
+        
+        If a new username cannot be saved, and error message will be sent to the user. At this point
+        manual login may be nessesary.
+        
+    Inputs: 
+        
+        hue_ip: the ip address of the hue bridge. Set by default to "10.88.21.10"
+        
+        file_path: the file name of the text file that will be saved containing the current username
+        
+    Outputs:
+        
+        b: bridge connection
+        
+    """
+    
+    if not path.exists(file_path):
+        
+        while True:
```

Comment:
This `while` loop is just going to repeat the "Cannot create new username" message over and over. Is the idea that it does that until someone presses the button? If so there is no way to break from this if no one can press the button. We'll want to rethink this based on what is wanted, but I would just remove the `while` for now and `return` in the `except` since the whole automatic login cannot be performed.